pkg/sql/distsqlrun/flow.go

```diff
@@ -47,6 +48,8 @@ type FlowID struct {
 type FlowCtx struct {
 	log.AmbientContext
 
+	stopper *stop.Stopper
```

Comment:
Ah, missed that link. This seems fine although I don't know this part of the code too well. Have you considered putting it in the `outbox`?